meson.build

```diff
@@ -0,0 +1,216 @@
+project(
+    'cloud-init',
+    version: run_command('./tools/read-version', check: true).stdout().strip(),
+    meson_version: '>=0.63.0',
+    default_options: [
+      # The default can yield broken results.
+      'python.install_env=auto'
+    ]
+)
+completions = dependency('bash-completion')
+systemd = dependency('systemd')
+system = host_machine.system()
+
+libdir = get_option('libdir')
+pkgdatadir = get_option('datadir')
+sysconfdir = get_option('sysconfdir')
+INIT_SYSTEM = get_option('init_system')
+VALID_INIT_SYSTEMS = [
+  'systemd', 'sysvinit', 'sysvinit_deb', 'sysvinit_dragonfly',
+  'sysvinit_freebsd', 'sysvinit_netbsd', 'sysvinit_openbsd'
+]
+if not VALID_INIT_SYSTEMS.contains(INIT_SYSTEM)
+  error(
+    'Invalid init_system supplied: "@0@". Must be one of: @1@'.format(
+      INIT_SYSTEM, VALID_INIT_SYSTEMS)
+  )
+endif
+
+
+DISTRO = get_option('distro')
+BSD_DISTROS = ['dragonfly', 'freebsd', 'netbsd', 'openbsd']
+RHEL_DISTROS = [
+  'almalinux', 'centos', 'cloudlinux', 'eurolinux', 'fedora', 'miraclelinux',
+  'rhel', 'rocky', 'virtuozzo'
+]
+VALID_DISTROS = [
+  'alpine', 'amazon', 'debian', 'mariner', 'opencloudos', 'openeuler',
+  'photon', 'ubuntu', 'suse', ''
+] + RHEL_DISTROS + BSD_DISTROS
+
+if not VALID_DISTROS.contains(DISTRO)
+  error(
+    'Invalid distro supplied: "@0@". Must be one of: @1@'.format(
+      DISTRO, VALID_DISTROS)
+  )
+endif
+
+
+if BSD_DISTROS.contains(system) or BSD_DISTROS.contains(DISTRO)
+  if INIT_SYSTEM = ''
+      if DISTRO == ''
+          DISTRO = system
+      endif
+      INIT_SYSTEM = 'sysvinit_' + DISTRO
+  endif
+else
+  if INIT_SYSTEM == ''
+    INIT_SYSTEM = 'systemd'
+  endif
+  if DISTRO == ''
+    fs = import('fs')
+    if fs.is_file('/etc/redhat-release')
+      message('Setting distro to rhel because of host /etc/redhat-release')
+      DISTRO = 'rhel'
+    endif
+  endif
+endif
+
+if RHEL_DISTROS.contains(DISTRO)
+  lib_exec_dir = join_paths('/usr', get_option('libexecdir'), 'cloud-init')
+else
+  lib_exec_dir = join_paths('/usr', 'lib', 'cloud-init')
+endif
+message('LIBEXEC: @0@ @1@'.format(lib_exec_dir, DISTRO))
+
+
+pymod = import('python')
+python = pymod.find_installation('python3', required: true)
+python.dependency()
```

Comment:
Why this? We don't use the returned [python dependency](https://mesonbuild.com/Python-module.html#python_dependency-object).